packages/functional-tests/src/tests/animation-scale-test.ts

```diff
@@ -0,0 +1,100 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import * as MRE from '@microsoft/mixed-reality-extension-sdk';
+
+import { Test } from '../test';
+
+const BounceHeightKeyframes: Array<MRE.Keyframe<number>> = [
+	{ time: 0, value: 0 },
+	{ time: 0.5, value: 1.1 },
+	{ time: 0.85, value: 0.1, easing: MRE.AnimationEaseCurves.EaseInQuadratic },
+	{ time: 1, value: 0 }];
+const BounceRotKeyframes: Array<MRE.Keyframe<MRE.Quaternion>> = [
+	{ time: 0, value: MRE.Quaternion.Identity() },
+	{ time: 0.33, value: MRE.Quaternion.FromEulerAngles(0, 2 * Math.PI / 3, 0) },
+	{ time: 0.67, value: MRE.Quaternion.FromEulerAngles(0, 4 * Math.PI / 3, 0) },
+	{ time: 1, value: MRE.Quaternion.Identity() }];
+const BounceScaleKeyframes: Array<MRE.Keyframe<MRE.Vector3>> = [
+	{ time: 0, value: { x: 1.33, y: 0.667, z: 1.33 }, easing: MRE.AnimationEaseCurves.EaseInQuadratic },
+	{ time: 0.15, value: { x: 0.667, y: 1.33, z: 0.667 } },
+	{ time: 0.85, value: { x: 0.667, y: 1.33, z: 0.667 }, easing: MRE.AnimationEaseCurves.Step },
+	{ time: 1, value: { x: 1.33, y: 0.667, z: 1.33 } },
+];
+
+export default class AnimationNativeTest extends Test {
```

Comment:
I'm seeing around 213 calls to GC.Alloc when this animation plays. That still seems like a lot.

Reply:
It's a lot, but I believe it's the minimum for this approach. My deep profile says it calls GC.Alloc once per animation, once per animation track, and once each time a JSON field is converted to a float during patching (Json.Net stores values in `object`s, so that's boxing and unboxing penalties). That totals to 7.0KB per frame.
![image](https://user-images.githubusercontent.com/1882376/79386221-ac9df180-7f1e-11ea-921d-93f8943cf678.png)

If this is too much we can discuss alternative strategies, but the only alternative that comes to mind is much more complicated: a fancy JSON parser that can separate tracks into value types and object types, and interpolate them differently at runtime. If it's even possible, it would take much longer to get right than this implementation.

It's worth noting that this probably is the worst-case scenario, animating 100 numeric fields (position, rotation, and scale on 10 objects). That's probably the total budget for MRE animations in a scene I'd guess.